mithril-signer/src/main.rs

```diff
@@ -236,6 +239,10 @@ async fn main() -> StdResult<()> {
         .send(())
         .map_err(|e| anyhow!("Metrics server shutdown signal could not be sent: {e:?}"))?;
 
+    if !preload_task.is_finished() {
```

Comment:
Could we add the `preload_task` to the `join_set` instead?

Reply:
No, that's the error that we did first.

The `join_set` only hold infinite tasks or tasks that wait for a signal (like `ctrl+c`), doing so allow us to wait for the first task of to finish and then to stop all other tasks.
So if we add the `preload_task` to the `join_set` it makes the note stop immediately after the preload_task finish, that's why it must not be included in the `join_set`.